p2p/host/autorelay/relay_finder.go

```diff
@@ -128,6 +142,15 @@ func (rf *relayFinder) background(ctx context.Context) {
 	oldCandidateTicker := rf.conf.clock.Ticker(rf.conf.maxCandidateAge / 5)
 	defer oldCandidateTicker.Stop()
 
+	scheduledWork := &scheduledWorkTimes{
+		nextRefresh:                 rf.conf.clock.Now().Add(rsvpRefreshInterval),
+		nextBackoff:                 rf.conf.clock.Now().Add(rf.conf.backoff / 5),
+		nextOldCandidateCheck:       rf.conf.clock.Now().Add(rf.conf.maxCandidateAge / 5),
+		nextAllowedCallToPeerSource: rf.conf.clock.Now().Add(-time.Second), // allow immediately
+		peerSourceTimer:             peerSourceTimer,
+		needMorePeers:               true,
```

Comment:
Having this bool feels a bit wonky, as it's used to communicate state between this function `runScheduledWork`. I think I'd prefer to
1. Make this a member variable.
1. Pass this struct by value to make it clear that no modification is happening.

Reply:
I don't want to make it a member variable because I don't want any other method to access it. It would be tempting to have two goroutines try to modify this and then you introduce a lock for what should be a scoped var. I also don't want to pollute the struct with this field that only this scope uses.

Modification is happening. All the next checks are replaced. We can have the function return a scheduledWorkTimes, not sure if that's much better though. wdyt?

Reply:
I'll refactor this differently. Coming soon :)

Reply:
Refactored this away.